### DISCUSSION TITLE
Re-exporting a UI library package to leverage `use client`

### DISCUSSION THREAD

**Discussion by milovangudelj** [vercel/next.js#46976] [Help]
### Summary

I'm trying to re-export a UI library package that doesn't make use of the `use client` directive. Specifically I would like to have a component, let's call it `UseClient`, which I can import and place somewhere on the page and access all the other UI components as sub-components like so

```jsx
<UseClient.Button>A button<UseClient.Button>
```

Right now `UseClient.tsx` looks like this

```jsx
"use client";

export * as UseClient from "@library/ui";
```

The editor is showing all the component types correctly and I'm able to get auto completion for each component's props. The issue is that when I try to run the development server it crashes with this error message

```terminal
error - TypeError: Cannot read properties of undefined (reading 'Button')
```

Why is that? What is it that I'm doing wrong here?

### Additional information

Here `@library/ui` is a package I'm building that integrates an icon library which doesn't yet make use of the `use client` directive. 

I'm still trying to figure out how to properly specify that a component should be a client component using tsup. So for the moment I'm still unable to slap `use client` wherever I need it and be done with it like that.

If you have suggestions on how to solve the problem using tsup directly that too would be appreciated.

**Comment by icyJoseph** [comment]
The problem here is with barrel files. https://github.com/vercel/next.js/issues/46293, not necessarily something you are doing wrong.

- https://github.com/vercel/next.js/issues/46293:

> Using export * inside a client boundary (a file with “use client”) is not currently supported, and an error should be displayed on the CLI. So this is a planned thing to work on rather than a bug.
